### DISCUSSION TITLE
Adobe Air contributes of android and ios application production

### DISCUSSION THREAD

**Discussion by hamraee** [airsdk/Adobe-Runtime-Support#2440] [General]
Do you know how much Adobe Air contributes to the production of Android and iOS applications?

**Comment by raresn** [comment]
Would be cool to know some statistics about AIR in general :)
I for one am curious how big the community is.

**Comment by 42retfa** [comment]
Hello,

I myself  would  be very interested in information on this subject

**Comment by ajwfrost** [comment]
We took a snapshot a little while ago of some of the ADT usage data, it had some interesting figures... (I can't remember what period of time it involved, we were processing it in Excel so just took 1m rows..)

Grouping via "application ID", it seems we have three times as many "free tier" applications as we do the commercially licensed ones; almost 50,000 different application IDs. But when I look at the number of transactions from the license database, the number of active commercial licensees is a lot less than this! so we perhaps have a lot of developers who have lots of application IDs. We don't actually track the number of downloads of the SDK (well, we can see the numbers and recently had logging going for a while because the data charges were surprisingly high -> turns out some people appear to download the SDKs repeatedly....)

From my management's view, the key thing is that (a) we cover the existing development team with the income, and (b) the income isn't going down. That last part is one metric that I'm particularly keen on - it's nice to see that we appear to have arrested the decline in terms of people moving away from AIR. I do wish we'd managed to pick it up perhaps a year earlier than we did, I think a number of studios switched away from AIR due to the period of silence from Adobe about its future.

Hope that's useful/interesting. Obviously the exact figures are somewhat commercially confidential. In terms of the 'community' it's definitely got a core of folk who stay fairly involved in these discussions/issues, and a lot of other people who use the AIR SDK without raising issues or posting here.

thanks

**Reply by jaycverg** [reply]
From our side, we have 1 desktop (Mac and Windows) and 1 android app (enterprise app used in warehouse inventory management) that are more than a decade old codebases. Moving away is not option for us when Adobe abandoned AIR. When Adobe announced that they will discontinue AIR, we had a small team started with the React Native project but because of the complexity of our app, it did not even get into 10% of the functionalities, then the devs moved out. 😄 

We are hoping that Harman will continue supporting AIR until it can.

**Comment by 42retfa** [comment]
Same here, we have one desktop application (windows and macOS) and one mobile application (Android an iOS) and switching to another technology is not an option since those 2 apps have large codebases. We are very happy to pay for the Harman Licence every year considering the way the Harman team is handling AIR  !

**Comment by hamraee** [comment]
I found this from appbrain.com:
[Link](https://www.appbrain.com/stats/libraries/details/air/adobe-air)
